db/dump.py

```diff
@@ -264,7 +387,13 @@ def import_db_dump(archive_path):
                         logging.info("Schema version verified.")
 
                 else:
-                    if file_name in table_names:
+                    if _is_partitioned_table_dump_file(file_name):
+                        table_name = member.name.split("/")[2]
```

Comment:
I think this is a little opaque and could be made more explicit. it looks like you're doing this:
 - If filename is lowlevel-1 then it's partitioned
 - member name is archivename/abdump/lowlevel/lowlevel-1, so the table name is member.split('/')[2] (lowlevel)

It'd be nice to either have a comment here explaining that partitioned tables are in a folder named by the table, and this is how we get it. This could be as simple as 
`archive, _, tablename, filename = member.name.split('/')`, so that it's explicit that the name should have 4 components, and what each part is.